lib/memory.c

```diff
@@ -124,33 +227,84 @@ M0_INTERNAL void *m0_alloc_nz(size_t size)
 }
 
 void *m0_alloc(size_t size)
+{
+	return alloc0(size);
+}
+
+void *(m0_alloc_profiled)(size_t size)
+{
+	return alloc0(size);
+}
+
+void m0_free(void *data)
+{
+	return free0(data);
+}
+
+M0_INTERNAL void *m0_alloc_aligned(size_t size, unsigned shift)
+{
+	return alloc0_aligned(size, shift);
+}
+
+M0_INTERNAL void *(m0_alloc_aligned_profiled)(size_t size, unsigned shift)
+{
+	return alloc0_aligned(size, shift);
+}
+
+M0_INTERNAL void m0_free_aligned(void *data, size_t size, unsigned shift)
+{
+	return free0_aligned(data, size, shift);
+}
+
+#endif
+
+M0_EXPORTED(m0_alloc_profiled);
+M0_EXPORTED(m0_alloc);
+M0_EXPORTED(m0_free);
+M0_EXPORTED(m0_alloc_aligned_profiled);
+M0_EXPORTED(m0_alloc_aligned);
+M0_EXPORTED(m0_free_aligned);
+
+static void *alloc0(size_t size)
 {
 	void *area;
 
         /* 9% of logs in m0trace log file is due to M0_ENTRY  and M0_LEAVE */
-	//M0_ENTRY("size=%zi", size);
-	if (M0_FI_ENABLED("fail_allocation"))
+#ifdef ENABLE_DEV_MODE
+	M0_ENTRY("size = %zi", size);
+#endif
+	if (M0_FI_ENABLED_IN("m0_alloc", "fail_allocation"))
 		return NULL;
 	area = m0_arch_alloc(size);
 	alloc_tail(area, size);
 	if (area != NULL) {
 		m0_arch_allocated_zero(area, size);
-	} else if (!M0_FI_ENABLED("keep_quiet")) {
+	} else {
+#ifdef ENABLE_DEV_MODE
+		if (!M0_FI_ENABLED_IN("m0_alloc", "keep_quiet")) {
+			M0_LOG(M0_ERROR, "Failed to allocate %zi bytes.", size);
+			m0_backtrace();
+		}
+#else
 		M0_LOG(M0_ERROR, "Failed to allocate %zi bytes.", size);
 		m0_backtrace();
+#endif
 	}
-	//M0_LEAVE("ptr=%p size=%zi", area, size);
+#ifdef ENABLE_DEV_MODE
```

Comment:
Same as above, please whenever possible try to avoid adding #ifdefs, they are ugly and difficult to understand and maintain.